lib/data/external/AwsClient.js

```diff
@@ -359,26 +359,21 @@ class AwsClient {
             return callback();
         });
     }
-    copyObject(request, sourceBucketName, sourceObjectKey, sourceLocation,
-    log, callback) {
+    copyObject(request, sourceKey, sourceLocation, log, callback) {
         const destBucketName = request.bucketName;
         const destObjectKey = request.objectKey;
         const destAwsKey = this._createAwsKey(destBucketName, destObjectKey,
           this._bucketMatch);
 
-        const sourceBucketMatch =
-        config.locationConstraints[sourceLocation].details.bucketMatch;
```

Comment:
do you not need this because the key already has the info?

Reply:
yes :)